SLHCUpgradeSimulations/Configuration/python/phase1TkCustoms.py

```diff
@@ -416,9 +428,10 @@ def customise_Reco(process,pileup):
     process.pixelTracks.FilterPSet.chi2 = cms.double(50.0)
     process.pixelTracks.FilterPSet.tipMax = cms.double(0.05)
     process.pixelTracks.RegionFactoryPSet.RegionPSet.originRadius =  cms.double(0.02)
-    process.templates.DoLorentz=False
-    process.templates.LoadTemplatesFromDB = cms.bool(False)
-    process.PixelCPEGenericESProducer.useLAWidthFromDB = cms.bool(False)
+    # use defaults d.k. 2/16
+    #process.templates.DoLorentz=False
+    #process.templates.LoadTemplatesFromDB = cms.bool(False)
+    #process.PixelCPEGenericESProducer.useLAWidthFromDB = cms.bool(False)

```

Comment:
Thanks Danek. Some (easy) parts of the customize were migrated to era (but in practice not used yet) during last autumn by Mark Grimes. My PR #12806 (integrated week ago, i.e. quite recently) started to use the `phase1Pixel` era along with the customize.

> I am not sure how this works, I assume :
> 1) everything in block “SiPixelSimBlock” is taken as default ?

Yes. The defaults should point to Run1.

> 2) things which are in  the “def” sections override the default, so I only need to put there things which are different?

Yes, exactly.

> 3) What is the difference between Phase1 and Phase1WithPileup?

If I understand the python code correctly, both functions get called when `phase1Pixel` era is active. Their Difference is that the Phase1 operates on SiPixelSimBlock, while Phase1WithPileup operates on the `cms.Process` object, and it only does something if `process.mix` exists and has been configured for greater-than-zero pileup.

> 4) How do I test it? I mean how do I switch between different “def” sections?

A full phase1 workflow can be run (e.g. in 800pre6) with `runTheMatrix.py -w upgrade -l 10000` (for four-muon gun). The generated configuration files have something like this
```
process = cms.Process('DIGI2RAW',eras.Run2_2017)
```
in their beginning. The second parameter specifies the active (set of) era(s) that are defined in https://github.com/cms-sw/cmssw/blob/CMSSW_8_0_X/Configuration/StandardSequences/python/Eras.py. So `Run2_2017` includes `phase1Pixel`, e.g. `Run2_25ns` includes `run2_25ns_specific`, and `Run2_50ns` includes `run2_50ns_specific`. The era can also be given as a command-line parameter to `cmsDriver.py`, but for testing just modifying the `Process` parameter is probably simpler.


> For the default part I would suggest to empty the “DeadModules” part.
> It is useless.

Personally I'm in favor of removing unused stuff, but this case is not my call.

> I would still like to keep the modified customs since some people might be using them.

That's something I'd also like to understand, i.e. what are the constraints in modifying/removing parts of `phase1TkCustoms.py`. In #12806 I also tried to keep it functional, but e.g. any tracking updates will be done with eras (it is actually pending only an agreement on how to do it), and after that it should be straightforward to migrate the rest of the customize to eras. At that point, if `phase1TkCustoms.py` is needed only for phase1 (and not for phase2), I would just remove the file.